SYCL/Basic/buffer/buffer.cpp

```diff
@@ -1,3 +1,4 @@
+// REQUIRES: aspect-fp64
```

Comment:
It doesn't look correct to me for this test and many others.
We probably have only a small portion of test that requires aspect-fp64. In such case we should split the test and have buffer.cpp without this requirement and buffer-aspect-fp64 for the part where it's required.

@intel/llvm-reviewers-runtime should review carefully.

Reply:
To split `buffer.cpp` to `buffer.cpp` and `buffer-aspect-fp64.cpp` with `// REQUIRES: aspect-fp64` is more reasonable.  But to split all of them (up to 50 test files), I need some time and effort to do this.

Or we may update test source with `device.has(sycl::aspect::fp64)` or other feature checkers to strip type coverage from test logic?

Reply:
I'm not sure what's the current state of device.has() support. I'm not sure if we have everything needed to mark code portion within a single source file. That would make sense if supported.

Reply:
If no objection, I will try to split 'double' type code to new `.cpp` file with `-aspect-fp64` suffix, and will create separate PRs for different code owner review.

Reply:
Revert `aspect-fp64` related changes, only keep Intel GPU FAMILY availability for restricting ESIMD raw_send tests.